gui/WorldView.cxx

```diff
@@ -1035,13 +1087,40 @@ void WorldView::setMesh(vtkSmartPointer<vtkPolyData> mesh)
 
   // Add this actor to the renderer
   d->renderer->AddActor(d->volumeActor);
-
-  if (mesh->GetPointData()->HasArray("RGB"))
+  bool rgb = false, rgba = false;
+  const char* arrayName = nullptr;
+  const char* rgbName = "RGB";
+  const char* rgbaName = "RGBA";
+  if (mesh->GetPointData()->HasArray(rgbName))
   {
+    rgb = true;
+    arrayName = rgbName;
+  }
+  if (mesh->GetPointData()->HasArray(rgbaName))
+  {
+    rgba = true;
+    arrayName = rgbaName;
+  }
+  if (rgb || rgba)
+  {
+    meshMapper->SetScalarModeToUsePointFieldData();
+    meshMapper->SelectColorArray(arrayName);
     meshMapper->ScalarVisibilityOn();
-    meshMapper->SelectColorArray("RGB");
     d->volumeOptions->setSurfaceColored(true);
   }
+  else if (mesh->GetPointData()->GetScalars())
+  {
+    // map active scalar
+    vtkDataArray* scalars = mesh->GetPointData()->GetScalars();
+    double range[2];
+    scalars->GetRange(range);
```

Comment:
```suggestion
    double range[2];
    mesh->GetPointData()->GetScalars()->GetRange(range);
```

Reply:
Why is this better?

Reply:
Oh, scalars is not used anywhere else. Sounds good.